index.bs

```diff
@@ -6583,6 +6737,360 @@ This extension enables use of a user verification method.
+
+    1. Let |dpkSig| be the result of signing the [=assertion signature=] [input](#fig-signature) with |devicePrivateKey|.
+
+    1. Output |dpkSig| as the extension's [=unsigned extension output=].
+
+        Note: |dpkSig| cannot be included in the [=authenticator extension output=] because it is returned inside the [=authenticator data=] and that would imply that the signature signs over itself.
+
+##### Attestation calculations ##### {#sctn-device-publickey-attestation-calculations}
+
+When computing attestations, the process in [[#sctn-generating-an-attestation-object]] takes two inputs: `authData` and `hash`. When calculating an attestation for a [=device-bound key=], the typical value for `hash` hashes over the attestation signature itself, which is impossible. Also the attestation of a [=device-bound key=] is potentially used repeatedly, thus may want to be cached. But signing over values that include [=[RP]=]-chosen nonces, like the [=hash of the serialized client data=], make that impossible.
+
+Therefore when calculating an attestation for a [=device-bound key=], the inputs are:
+
+    * For `authData`, substitute the concatenation of the byte string h'64657669636520626f756e64206b6579206174746573746174696f6e2073696700ffffffff' and the value of |aaguid| from the extension output.
+    * For `hash`, substitute the concatenation of the |dpk| and |nonce| fields from the extension output. (The nonce may be empty.)
+
+The attestation signature is thus typically calculated over the bytes of <code>(h'64657669636520626f756e64206b6579206174746573746174696f6e2073696700ffffffff' || |aaguid| || |dpk| || |nonce|)</code>. The 37-byte prefix ensures domain separation: it takes the place of the RP ID hash, flags, and signature counter fields in those messages and ensures that no attestation signature for a [=device-bound key=] can be confused with a signature for a [=user credential=].
+
+Note that when |nonce| is empty, then the (signed) authenticator extension output MAY be constant. However, the (unsigned) |dpkSig| output is always unique and prevents replay of the (signed) extension output without knowledge of the [=device private key=].
+
+#### `devicePubKey` Extension Output Verification Procedures #### {#sctn-device-publickey-extension-verification}
+
+Verifying the <code>[=devicePubKey=]</code> extension output is performed by the [=[RP]=] whenever a <i>new</i> [=device public key=] is returned within the extension output. As explained in [[#sctn-device-publickey-extension-usage]], a new [=device public key=] is always returned as a result of a {{CredentialsContainer/create()|navigator.credentials.create()}} call (i.e., as part of a [=registration ceremony=]). In contrast, a new [=device public key=] <i>may</i> be returned as a result of a {{CredentialsContainer/get()|navigator.credentials.get()}} call (i.e., as part of an [=authentication ceremony=]).
```

Comment:
>As explained in [§ 10.2.2.1 Relying Party Usage](https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1663.html#sctn-device-publickey-extension-usage), a new [device public key](https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1663.html#device-public-key) is always returned as a result of a [navigator.credentials.create()](https://w3c.github.io/webappsec-credential-management/#dom-credentialscontainer-create) call (i.e., as part of a [registration ceremony](https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1663.html#registration-ceremony)). In contrast, a new [device public key](https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1663.html#device-public-key) may be returned as a result of a [navigator.credentials.get()](https://w3c.github.io/webappsec-credential-management/#dom-credentialscontainer-get) call (i.e., as part of an [authentication ceremony](https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1663.html#authentication-ceremony)).

The second sentence reads a bit weird - the contrast isn't very clear, it's just saying both cases (might) create a new key. Perhaps better something like this?

>In contrast, a navigator.credentials.get() call (i.e., as part of an authentication ceremony) MAY create and return new device public key, or MAY reuse an existing one.

Reply:
Done.